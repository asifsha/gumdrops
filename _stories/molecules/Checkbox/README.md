The `<Checkbox>` component can be wrapped with the `<FormGroup>` component. It can be used as a controlled or uncontrolled component.

**NOTE**: This component can be controlled or uncontrolled. This component accepts any prop that you may want to add, which are applied to the input. The only drawback is that there is not validation for unlisted props. **It is highly recommended that you use controlled components in React.**

*Recommended Props*: `name`, `checked`, `defaultChecked`, `onChange`

**Controlled example - Example with reusable checkbox handler**:
```
state {
    parrots: false
}
_handleCheckboxChange = ({ target }) => {
    const { name, checked } = target;

    this.setState({
        [name]: checked
    });
}
return(
    <FormGroup>
        <Checkbox
            name="parrots"
            label="party parrots"
            checked={ this.state.parrots }
            onChange={ this._handleCheckboxChange }
        />
    </FormGroup>
)
```


**Uncontrolled example**:
```
_handleSubmit = (e) => {
    e.preventDefault();
    const {
        parrots,
        otherName,
        otherName2
    } = e.target.elements;
    console.log(parrots.checked);
};

render() {
    return (
        <form onSubmit={ this._handleSubmit }>
            <FormGroup>
                { /* Pass defaultChecked if you want
                the input to be checked initially */ }
                <Checkbox
                    name="parrots"
                    label="party parrots"
                    defaultChecked
                />
            </FormGroup>
        </form>
    );
}
```
