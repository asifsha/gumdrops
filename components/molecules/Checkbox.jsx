import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Checkbox = ({ label, size, style, className, ...otherProps }) => {
    const baseClass = 'gds-form-group__checkbox';
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${size}`]: size
    });

    return (
        <div className={rootClass} style={style}>
            <label className="gds-form-group__checkbox-label">
                <input className="gds-form-group__checkbox-input" type="checkbox" {...otherProps} />
                <span className="gds-form-group__checkbox-indicator" />
                {label}
            </label>
        </div>
    );
};

Checkbox.displayName = 'Checkbox';

Checkbox.propTypes = {
    label: PropTypes.string,
    /** xs, sm */
    size: PropTypes.oneOf(['xs', 'sm']),
    className: PropTypes.string,
    style: PropTypes.object
};

export default Checkbox;
