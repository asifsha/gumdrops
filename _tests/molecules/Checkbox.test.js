/* globals mount */
import React from 'react';
import Checkbox from '../../components/molecules/Checkbox';

const defaultProps = {
    label: 'Check Me',
    size: 'xs',
    className: 'custom-class',
    style: { width: 140 }
};

describe('Expect <Checkbox>', () => {
    it('to render', () => {
        const wrapper = mount(<Checkbox {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render with other props', () => {
        const props = {
            ...defaultProps,
            onChange: () => {},
            value: 'foo'
        };
        const wrapper = mount(<Checkbox {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
