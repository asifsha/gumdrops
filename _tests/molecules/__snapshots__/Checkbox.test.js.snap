// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Checkbox> to render 1`] = `
<Checkbox
  className="custom-class"
  label="Check Me"
  size="xs"
  style={
    Object {
      "width": 140,
    }
  }
>
  <div
    className="gds-form-group__checkbox custom-class gds-form-group__checkbox--xs"
    style={
      Object {
        "width": 140,
      }
    }
  >
    <label
      className="gds-form-group__checkbox-label"
    >
      <input
        className="gds-form-group__checkbox-input"
        type="checkbox"
      />
      <span
        className="gds-form-group__checkbox-indicator"
      />
      Check Me
    </label>
  </div>
</Checkbox>
`;

exports[`Expect <Checkbox> to render with other props 1`] = `
<Checkbox
  className="custom-class"
  label="Check Me"
  onChange={[Function]}
  size="xs"
  style={
    Object {
      "width": 140,
    }
  }
  value="foo"
>
  <div
    className="gds-form-group__checkbox custom-class gds-form-group__checkbox--xs"
    style={
      Object {
        "width": 140,
      }
    }
  >
    <label
      className="gds-form-group__checkbox-label"
    >
      <input
        className="gds-form-group__checkbox-input"
        onChange={[Function]}
        type="checkbox"
        value="foo"
      />
      <span
        className="gds-form-group__checkbox-indicator"
      />
      Check Me
    </label>
  </div>
</Checkbox>
`;
